auditloggingccl: TestRoleBasedAuditEnterpriseGated is flaky

First observed this failure in a PR that changed unrelated code [1]. I was also able to reproduce the failure by stressing the test with the command below:

```console
$ ./dev test ./pkg/ccl/auditloggingccl/ -f TestRoleBasedAuditEnterpriseGated --stress --ignore-cache --cpus 16 --timeout 2m
...
I230628 18:49:36.462093 1 (gostd) rand.go:249  [-] 1  random seed: 1654522793035659324
test logs left over in: /tmp/cockroach/_tmp/615fe16b7e538334224f307139f5bdb8/logTestRoleBasedAuditEnterpriseGated1310055266
--- FAIL: TestRoleBasedAuditEnterpriseGated (1.28s)
    test_log_scope.go:167: test logs captured to: /tmp/cockroach/_tmp/615fe16b7e538334224f307139f5bdb8/logTestRoleBasedAuditEnterpriseGated1310055266
    audit_logging_test.go:108: enterprise is enabled, expected 1 entry, got 2
    panic.go:522: -- test log scope end --
FAIL
```

[1] <https://teamcity.cockroachdb.com/viewLog.html?buildId=10715979&buildTypeId=Cockroach_BazelEssentialCi>

Jira issue: CRDB-29173

--- a/pkg/ccl/auditloggingccl/audit_logging_test.go
+++ b/pkg/ccl/auditloggingccl/audit_logging_test.go
@@ -95,7 +95,7 @@ func TestRoleBasedAuditEnterpriseGated(t *testing.T) {
 		0,
 		math.MaxInt64,
 		10000,
-		regexp.MustCompile(`"EventType":"role_based_audit_event"`),
+		regexp.MustCompile(`"Statement":"SHOW CLUSTER SETTING \\"sql.log.user_audit\\""`),
 		log.WithMarkedSensitiveData,
 	)
 

